hoomd/mpcd/SphereGeometry.h

```diff
@@ -85,25 +107,31 @@ class __attribute__((visibility("default"))) SphereGeometry
              * The perpendicular and parallel components of the velocity are:
              * v_perp = (v.n)n = (v.r/R^2)r
              * v_para = v-v_perp
+	     * V_vec is vector component of V(interface velocity)
              */
+	    const Scalar3 V_vec = m_V*pos/(m_R-m_V*dt);
+	    const Scalar3 vperp = (dot(vel,pos)/m_R2)*pos;
+
+
             if (m_bc == boundary::no_slip)
                 {
-                /* No-slip and no penetration requires reflection of both parallel and perpendicular components.
-                 * This results in just flipping of all the velocity components.
+                /* No-slip and no penetration requires reflection of both parallel component and perpendicular(relative to interface) component
+		 * V_perp(new) = -(V_perp(old)-V_interface)+V_interface 
+                 * V_para(new) = -V_para(old)
+                 * This results in just V_new = - v_old + 2*V_interface. 
                  */
-                vel = -vel;
+                vel = -vel + Scalar(2)*V_vec;
                 }
             else if (m_bc == boundary::slip)
                 {
                 /*
-                 * Only no-penetration condition is enforced, so only v_perp is reflected.
+                 * Only no-penetration condition is enforced, so only v_perp(relative to interface) is reflected.
                  * The new velocity v' is:
-                 * v' = -v_perp + v_para = v - 2*v_perp
+                 * v' = v_old - 2*v_perp + 2*V_interface
                 */
-                const Scalar3 vperp = (dot(vel,pos)/m_R2)*pos;
-                vel -= Scalar(2)*vperp;
+                vel -= Scalar(2)*(vperp - V_vec);
                 }
-            return true;
+	    return true;
```

Comment:
Thank you, I now noticed that since I was working with different text editor from my laptop(which had different tab settings) , many of the indentation got wrong , gonna fix that.